pkg/service/service.go

```diff
@@ -1,13 +1,125 @@
 package service
 
 import (
+	"context"
+
 	runapi "github.com/GoogleCloudPlatform/cloud-run-release-operator/internal/run"
+	"github.com/GoogleCloudPlatform/cloud-run-release-operator/pkg/config"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/api/run/v1"
 )
 
-// Client is the connection to update and obtain data about a single service.
+// Client is the connection to update and obtain data about a single service. It
+// contains a client to the Cloud Run API, information about the service and the
+// service object.
 type Client struct {
 	RunClient   runapi.Client
+	Service     *run.Service
 	Project     string
 	ServiceName string
 	Region      string
```

Comment:
Given project, name, and region are all things that can be found on the service object, is there a value in duplicate data? If not, maybe these would be better served by helper methods.

Reply:
I'm not sure that all these data can be extracted from the service object. Looking at the YAML in the Cloud Run service, I get

```yaml
apiVersion: serving.knative.dev/v1
kind: Service
metadata:
  name: hello
  namespace: '929898821634'
  generation: 130
  creationTimestamp: '2020-06-29T22:14:02.259763Z'
  labels:
    rollout-strategy: gradual
    cloud.googleapis.com/location: us-east1
  annotations:
    run.googleapis.com/client-name: gcloud
    rollout.cloud.run/stableRevision: hello-00124-cim
    client.knative.dev/user-image: gcr.io/cloudrun/hello
    run.googleapis.com/client-version: 298.0.0
    run.googleapis.com/launch-stage: ALPHA
```
Can we trust that all types of deployment (not using gcloud, for instance) are going to include the `name` and `namespace` metadata? The region looks like a label that was created by my deployment tool (gcloud).

Reply:
name/namespace are mandatory.
however don't rely on Google Cloud specific annotations since we might support knative someday.

If you keep passing `<region,name,project>` triplet around, just create a `ServiceRecord` struct?